posthog/api/feature_flag.py

```diff
@@ -1134,6 +1135,86 @@ def _update_feature_flag_dashboard(feature_flag: FeatureFlag, old_key: str) -> N
     update_feature_flag_dashboard(feature_flag, old_key)
 
 
+class GroupsJSONField(serializers.CharField):
+    """
+    CharField that parses JSON object strings.
+    Matches legacy behavior of json.loads(request.GET.get("groups", "{}")).
+    """
+
+    def __init__(self, **kwargs):
+        kwargs.setdefault("required", False)
+        kwargs.setdefault("default", "{}")
+        kwargs.setdefault("allow_blank", True)
+        kwargs.setdefault("help_text", "Groups for feature flag evaluation (JSON object string)")
+        super().__init__(**kwargs)
+
+    def to_internal_value(self, data):
+        # Handle case where data is already a dict (from previous parsing or direct assignment)
+        if isinstance(data, dict):
+            return data
+
+        value = super().to_internal_value(data)
+        if not value:
+            return {}
+        try:
+            parsed = json.loads(value)
+            if not isinstance(parsed, dict):
+                raise serializers.ValidationError("groups must be a JSON object")
+            return parsed
+        except (json.JSONDecodeError, ValueError):
+            raise serializers.ValidationError("Invalid JSON in groups parameter")
+
+
+class MyFlagsQuerySerializer(serializers.Serializer):
+    groups = GroupsJSONField()
+
+
+class LocalEvaluationQuerySerializer(serializers.Serializer):
+    send_cohorts = serializers.BooleanField(
+        required=False, default=False, allow_null=True, help_text="Include cohorts in response"
+    )
+
+    def to_internal_value(self, data):
+        # Handle ?send_cohorts without a value (empty string) as True
+        if "send_cohorts" in data and data["send_cohorts"] == "":
+            data = data.copy()
+            data["send_cohorts"] = True
+        return super().to_internal_value(data)
+
+
+class EvaluationReasonsQuerySerializer(serializers.Serializer):
+    distinct_id = serializers.CharField(required=True, help_text="User distinct ID")
+    groups = GroupsJSONField()
+
+
+class ActivityQuerySerializer(serializers.Serializer):
+    limit = serializers.IntegerField(required=False, default=10, min_value=1, help_text="Number of items per page")
+    page = serializers.IntegerField(required=False, default=1, min_value=1, help_text="Page number")
+
+
+class EvaluationReasonSerializer(serializers.Serializer):
+    reason = serializers.CharField(help_text="The reason for the evaluation result")
+    condition_index = serializers.IntegerField(
+        allow_null=True, help_text="The index of the condition that matched, if applicable"
+    )
+
+
+class FlagEvaluationResultSerializer(serializers.Serializer):
+    value = serializers.JSONField(help_text="The evaluated value of the feature flag (boolean or variant key string)")
+    evaluation = EvaluationReasonSerializer()
```

Comment:
I see this referenced in a doc comment, but its not actually referenced anywhere. I wonder if you meant to use it in `EvaluationReasonsResponseSerializer`.